src/datasources/asset-utilization/helper.ts

```diff
@@ -0,0 +1,311 @@
+    endDate.setHours(endDate.getHours() - timeZoneOffset);
+
+    // Format the Date objects back to time strings
+    let startTime = startDate.toISOString().split('T')[1].substr(0, 8);
+    let endTime = endDate.toISOString().split('T')[1].substr(0, 8);
+
+    return {startTime, endTime};
+}
+
+export const extractTimestampesFromData = (history: AssetUtilizationHistory[]) => {
+    return history.map((item) => {
+        return {
+            'startTimestamp': new Date(item.startTimestamp).getTime(),
+            'endTimestamp': item.endTimestamp ? new Date(item.endTimestamp).getTime() : 0,
+            'heartbeatTimestamp': item.heartbeatTimestamp ? new Date(item.heartbeatTimestamp).getTime() : 0
+        }
+    })
+}
+
+export const sortByStartTimestamp = (intervals: Array<IntervalWithHeartbeat<number>>) => {
+    if (intervals.length <= 1) {
+        return intervals;
+    }
+
+    // Sort intervals based on the startTimestamp
+    intervals.sort((a, b) => a.startTimestamp - b.startTimestamp);
+
+    return intervals
+}
+export const patchMissingEndTimestamps = (history: Array<IntervalWithHeartbeat<number>>) => {
+    return history.map((item, index) => {
+        let newItem: Interval<number> = {
+            'startTimestamp': new Date(item.startTimestamp).getTime(),
+            'endTimestamp': 0
+        }
+
+        if (!item.endTimestamp) {
+            if (!item.heartbeatTimestamp) {
+                newItem.endTimestamp = new Date(item.startTimestamp).getTime() + 10 * minuteInSeconds
+            } else {
+                newItem.endTimestamp = new Date(item.heartbeatTimestamp).getTime()
+            }
+        } else {
+            newItem.endTimestamp = new Date(item.endTimestamp).getTime()
+        }
+        return newItem
+    })
+}
+
+export const filterDataByTimeRange = (data: Array<Interval<number>>, from: number, to: number) => {
+    return data.filter((interval: Interval<number>) => {
+        return interval.endTimestamp > from && interval.startTimestamp < to;
+    })
+}
+export const mergeOverlappingIntervals = (intervals: Array<Interval<number>>) => {
+    if (intervals.length <= 1) {
+        return intervals;
+    }
+
+    const mergedIntervals = [intervals[0]];
+
+    for (let i = 1; i < intervals.length; i++) {
+        const currentInterval = intervals[i];
+        const previousInterval = mergedIntervals[mergedIntervals.length - 1];
+
+        if (currentInterval.startTimestamp <= previousInterval.endTimestamp) {
+            // Overlapping intervals, merge them
+            mergedIntervals[mergedIntervals.length - 1].endTimestamp = Math.max(
+                previousInterval.endTimestamp,
+                currentInterval.endTimestamp
+            );
+        } else {
+            // Non-overlapping intervals, add the current interval to the result
+            mergedIntervals.push(currentInterval);
+        }
+    }
+
+    return mergedIntervals;
+}
+
+export const getBusinessHours = (
```

Comment:
This is a super long function, could it be split up?

Reply:
changed to another super long function with some description